pyproject.toml

```diff
@@ -54,20 +54,36 @@ classifiers=[
 ]
 
 [tool.poetry.dependencies]
-python = ">=3.9,<3.13"
-requests = "^2.31.0"
+python = ">=3.8,<3.13"
 typing-extensions = ">=3.7.4"
-python-dateutil = ">=2.5.3"
-urllib3 = "2.0.7"
+urllib3 = ">=1.26.16"
 tqdm = ">=4.64.1"
 grpcio = { version = ">=1.44.0", optional = true }
 grpc-gateway-protoc-gen-openapiv2 = { version = "0.1.0", optional = true }
 googleapis-common-protos = { version = ">=1.53.0", optional = true }
 lz4 = { version = ">=3.1.3", optional = true }
 protobuf = { version = "~=3.20.0", optional = true }
 
+[tool.poetry.group.types]
+optional = true
```

Comment:
Does this just flag everything in that group as optional rather than marking them explicitly?

Reply:
Poetry has "groups" for dev dependencies and "extras" for installable add-ons like "grpc". Similar concepts, but one is for us and one is for the end-users.

Groups are all included by default when you run `poetry install` but you can invert that behavior for specific groups by marking them optional and then explicitly controlling the groups you would like to install using `--with` and `--without` options. E.g. `poetry install --without dev` would install with nothing from the dev group (otherwise included by default), and `poetry install --with types` would install with the `dev` group (included by default) plus the `types` group (optional, so only included when specifically asked for).